cypress/integration/fileMenu.cy.js

```diff
@@ -0,0 +1,153 @@
+
+        assertItems({
+            [ITEM_NEW]: true,
+            [ITEM_OPEN]: true,
+        })
+    })
+
+    it('reflects unsaved state', () => {
+        cy.visit('/', EXTENDED_TIMEOUT)
+
+        selectEventProgram({
+            programName: 'Adverse events following immunization',
+        })
+
+        clickMenubarUpdateButton()
+
+        cy.getBySel('menubar').contains('File').click()
+
+        assertItems({
+            [ITEM_NEW]: true,
+            [ITEM_OPEN]: true,
+            [ITEM_SAVE]: true,
+        })
+    })
+
+    it('reflects saved state', () => {
+        cy.visit('/#/ZTrsv19jw9U', EXTENDED_TIMEOUT)
+
+        cy.getBySel('visualization-title').contains('COVAC enrollment')
+
+        cy.getBySel('menubar').contains('File').click()
+
+        assertItems({
+            [ITEM_NEW]: true,
+            [ITEM_OPEN]: true,
+            [ITEM_SAVEAS]: true,
+            [ITEM_RENAME]: true,
+            [ITEM_TRANSLATE]: true,
+            [ITEM_SHARING]: true,
+            [ITEM_GETLINK]: true,
+            [ITEM_DELETE]: true,
+        })
+    })
+
+    it('reflects dirty state (legacy: do not allow saving)', () => {
+        cy.visit('/#/ZTrsv19jw9U', EXTENDED_TIMEOUT)
+
+        cy.getBySel('visualization-title').contains('COVAC enrollment')
+
+        clickMenubarUpdateButton()
+
+        cy.getBySel('visualization-title').contains('Edited')
+
+        cy.getBySel('menubar').contains('File').click()
+
+        assertItems({
+            [ITEM_NEW]: true,
+            [ITEM_OPEN]: true,
+            [ITEM_SAVEAS]: true,
+            [ITEM_RENAME]: true,
+            [ITEM_TRANSLATE]: true,
+            [ITEM_SHARING]: true,
+            [ITEM_GETLINK]: true,
+            [ITEM_DELETE]: true,
+        })
+    })
+
+    it('reflects dirty state (new: created in this app)', () => {
+        cy.visit('/', EXTENDED_TIMEOUT)
+
+        selectEventProgram({
+            programName: 'Adverse events following immunization',
+        })
+
+        clickMenubarUpdateButton()
+
+        cy.getBySel('menubar').contains('File').click()
+
+        cy.getBySel(ITEM_SAVE).click()
+
+        const name = 'Cypress test dirty state'
+
+        typeInput('file-menu-saveas-modal-name-content', name)
+
+        cy.getBySel('file-menu-saveas-modal-save').click()
+
+        cy.getBySel('visualization-title').contains(name)
+
+        clickMenubarUpdateButton()
+
+        cy.getBySel('visualization-title').contains('Edited')
+
+        cy.getBySel('menubar').contains('File').click()
+
+        assertItems({
+            [ITEM_NEW]: true,
+            [ITEM_OPEN]: true,
+            [ITEM_SAVE]: true,
+            [ITEM_SAVEAS]: true,
+            [ITEM_RENAME]: true,
+            [ITEM_TRANSLATE]: true,
+            [ITEM_SHARING]: true,
+            [ITEM_GETLINK]: true,
+            [ITEM_DELETE]: true,
+        })
+
+        // cleanup
+        cy.getBySel(ITEM_DELETE).click()
+        cy.getBySel('file-menu-delete-modal-delete').click()
```

Comment:
I would add some sort of assert after this step to make sure that it was indeed deleted.
Either looking for the "deleted" message and/or run the `assertItems` again to make sure the file menu was reset to "default".
Otherwise the delete here might fail silently without us ever knowing (i.e. rule of thumb, always end all tests with an assertion).

Reply:
I'm thinking that the delete function should have it's own test, instead of "hiding" it in a disable/enable test, and to avoid overlapping we should not test it again here. But I guess it's okay to test the frontend part of it, i.e. that the menu gets reset.